src/C4MapCreatorS2.h

```diff
@@ -349,8 +349,11 @@ class C4MapCreatorS2 : public C4MCNode
 {
 public:
 	C4MapCreatorS2(C4SLandscape *pLandscape, C4TextureMap *pTexMap, C4MaterialMap *pMatMap, int iPlayerCount);
+	C4MapCreatorS2(C4MCNode *pOwner, C4MapCreatorS2 &rTemplate, C4SLandscape *pLandscape, bool fClone); // construct of template
 	~C4MapCreatorS2();
 
+	C4MapCreatorS2 *clone(C4MCNode *pToNode, C4SLandscape *pLandscape) { return new C4MapCreatorS2(pToNode, *this, pLandscape, true); }
```

Comment:
Is `pToNode` actually used? Seems like it is only ever used with `nullptr`. In that case please remove the argument from `clone` and the constructor.

Reply:
Instead of returning a pointer, return a `unique_ptr`.

Reply:
What is the rationale for having the `clone` function as well as the additional public constructor?

Reply:
> Is `pToNode` actually used? Seems like it is only ever used with `nullptr`. In that case please remove the argument from `clone` and the constructor.

> What is the rationale for having the `clone` function as well as the additional public constructor?

The function head was like this because it is a copy of the clone functions of other C4MCNodes. I'd have stated that it might be usefull to have it be similar as a user might not know what kind of node their creating but since the clone function for the MapCreator does require additional parameters, that's not the case. I'll remove it.